pinot-broker/src/main/java/com/linkedin/pinot/broker/broker/helix/BrokerResourceOnlineOfflineStateModelFactory.java

```diff
@@ -113,7 +127,12 @@ public void onBecomeDroppedFromOffline(Message message, NotificationContext cont
       try {
         LOGGER.info("BrokerResourceOnlineOfflineStateModel.onBecomeDroppedFromOffline() : " + message);
         String tableName = message.getPartitionName();
+        TableConfig tableConfig = ZKMetadataProvider.getTableConfig(_propertyStore, tableName);
+
         _helixExternalViewBasedRouting.markDataResourceOffline(tableName);
+        _tableQueryQuotaManager.dropTableQueryQuota(
+            tableConfig,
```

Comment:
Dropping the table query quota should really only depend on the table name, not any other thing (external view, table configuration). If there is no quota configuration, by definition, dropping the table query quota does nothing. This also ensures that `dropTableQueryQuota` will always behave as expected, even if there are odd edge cases where the table quota configuration is changed while the broker is running.